### ISSUE TITLE
Fluent-bit input plugin tail doesn't process all logs: scan_blog add(): dismissed:

### ISSUE THREAD

**Issue by rguske** [fluent/fluent-bit#4155]
## Bug Report

**Describe the bug**
Fluent Bit is not processing all logs located in `/var/log/containers/`. 

**To Reproduce**
The following messages are displayed:

```
[2021/10/01 14:40:05] [debug] [input:tail:tail.0] scanning path /var/log/containers/*.log                                                                                                                                                                                                                                               │
│ [2021/10/01 14:40:05] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /var/log/containers/activator-85cd6f6f9-nrncf_knative-serving_activator-3b631f27f6667599ae940f94afe6a65a4d1d488e7979fced513fa910082a5ae1.log, inode 404768                                                                                                │
│ [2021/10/01 14:40:05] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /var/log/containers/activator-85cd6f6f9-nrncf_knative-serving_activator-ca32320178170fe4198ce1b0bd57d8ea031c7c886a7b0e3d66bb1b78b67613b8.log, inode 921337                                                                                                │
│ [2021/10/01 14:40:05] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /var/log/containers/antrea-agent-gql5r_kube-system_antrea-agent-63659cdc8e5ddba3eaf729be280661b45fd198e6d2c7195965be85cdca81f41a.log, inode 536837                                                                                                        │
│ [2021/10/01 14:40:05] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /var/log/containers/antrea-agent-gql5r_kube-system_antrea-agent-8726abf73577f597e15716176cfcdce442b159d00ec12f59e439719d824a9585.log, inode 1190181                                                                                                       │
│ [2021/10/01 14:40:05] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /var/log/containers/antrea-agent-gql5r_kube-system_antrea-ovs-08045b767f2f8ee421b3b4d8d5b646b49b4e12199ae957cad178dd3d11670ec6.log, inode 663855 
```
- Steps to reproduce the problem:
Configuration details:

ServiceAccount:

```
rules:
- apiGroups:
  - ""
  resources:
  - namespaces
  - pods
  verbs:
  - get
  - list
  - watch
 ```

ConfigMap:

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: fluent-bit-config
  namespace: vmware-system
  labels:
    k8s-app: fluent-bit
apiVersion: v1
data:
  filter-kubernetes.conf: |
    [FILTER]
        Name                kubernetes
        Match               kube.*
        Kube_URL            https://kubernetes.default.svc:443
        Kube_CA_File        /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
        Kube_Token_File     /var/run/secrets/kubernetes.io/serviceaccount/token
        Kube_Tag_Prefix     kube.var.log.containers.
        Merge_Log           On
        Merge_Log_Key       log_processed
        K8S-Logging.Parser  On
        K8S-Logging.Exclude Off
    [FILTER]
        Name                  modify
        Match                 kube.*
        Copy                  kubernetes k8s

    [FILTER]
        Name                  nest
        Match                 kube.*
        Operation             lift
        Nested_Under          kubernetes
  filter-record.conf: |
    [FILTER]
        Name                record_modifier
        Match               *
        Record tkg_cluster veba-demo.jarvis.tanzu
        Record tkg_instance veba-demo.jarvis.tanzu
    [FILTER]
        Name                  nest
        Match                 kube.*
        Operation             nest
        Wildcard              tkg_instance*
        Nest_Under            tkg

    [FILTER]
        Name                  nest
        Match                 kube_systemd.*
        Operation             nest
        Wildcard              SYSTEMD*
        Nest_Under            systemd
  fluent-bit.conf: |
    [SERVICE]
        Flush         1
        Log_Level     debug
        Daemon        off
        Parsers_File  parsers.conf
        HTTP_Server   On
        HTTP_Listen   0.0.0.0
        HTTP_Port     2020

    @INCLUDE input-kubernetes.conf
    @INCLUDE input-systemd.conf
    @INCLUDE input-kube-apiserver.conf
    @INCLUDE input-auditd.conf
    @INCLUDE filter-kubernetes.conf
    @INCLUDE filter-record.conf
    @INCLUDE output-syslog.conf
  input-auditd.conf: |
    [INPUT]
        Name              tail
        Tag               audit.*
        Path              /var/log/audit/audit.log
        Parser            logfmt
        DB                /var/log/flb_system_audit.db
        Mem_Buf_Limit     50MB
        Refresh_Interval  10
        Skip_Long_Lines   On
  input-kube-apiserver.conf: |
    [INPUT]
        Name              tail
        Tag               apiserver_audit.*
        Path              /var/log/kubernetes/audit.log
        Parser            json
        DB                /var/log/flb_kube_audit.db
        Mem_Buf_Limit     50MB
        Refresh_Interval  10
        Skip_Long_Lines   On
  input-kubernetes.conf: |
    [INPUT]
        Name              tail
        Tag               kube.*
        Path              /var/log/containers/*.log
        Parser            docker
        DB                /var/log/flb_kube.db
        Mem_Buf_Limit     50MB
        Skip_Long_Lines   On
        Refresh_Interval  10
  input-systemd.conf: |
    [INPUT]
        Name                systemd
        Tag                 kube_systemd.*
        Path                /var/log/journal
        DB                  /var/log/flb_kube_systemd.db
        Systemd_Filter      _SYSTEMD_UNIT=kubelet.service
        Systemd_Filter      _SYSTEMD_UNIT=containerd.service
        Read_From_Tail      On
        Strip_Underscores   On
  output-syslog.conf: |
    [OUTPUT]
        Name   syslog
        Match  kube.*
        Host   10.197.79.57
        Port   514
        Mode   tcp
        Syslog_Format        rfc5424
        Syslog_Hostname_key  tkg_cluster
        Syslog_Appname_key   pod_name
        Syslog_Procid_key    container_name
        Syslog_Message_key   message
        Syslog_SD_key        k8s
        Syslog_SD_key        labels
        Syslog_SD_key        annotations
        Syslog_SD_key        tkg

    [OUTPUT]
        Name   syslog
        Match  kube_systemd.*
        Host   10.197.79.57
        Port   514
        Mode   tcp
        Syslog_Format        rfc5424
        Syslog_Hostname_key  tkg_cluster
        Syslog_Appname_key   tkg_instance
        Syslog_Message_key   MESSAGE
        Syslog_SD_key        systemd
  parsers.conf: |
    [PARSER]
        Name   apache
        Format regex
        Regex  ^(?<host>[^ ]*) [^ ]* (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^\"]*?)(?: +\S*)?)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>[^\"]*)" "(?<agent>[^\"]*)")?$
        Time_Key time
        Time_Format %d/%b/%Y:%H:%M:%S %z

    [PARSER]
        Name   apache2
        Format regex
        Regex  ^(?<host>[^ ]*) [^ ]* (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^ ]*) +\S*)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>[^\"]*)" "(?<agent>[^\"]*)")?$
        Time_Key time
        Time_Format %d/%b/%Y:%H:%M:%S %z

    [PARSER]
        Name   apache_error
        Format regex
        Regex  ^\[[^ ]* (?<time>[^\]]*)\] \[(?<level>[^\]]*)\](?: \[pid (?<pid>[^\]]*)\])?( \[client (?<client>[^\]]*)\])? (?<message>.*)$

    [PARSER]
        Name   nginx
        Format regex
        Regex ^(?<remote>[^ ]*) (?<host>[^ ]*) (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^\"]*?)(?: +\S*)?)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>[^\"]*)" "(?<agent>[^\"]*)")?$
        Time_Key time
        Time_Format %d/%b/%Y:%H:%M:%S %z

    [PARSER]
        Name   json
        Format json
        Time_Key time
        Time_Format %d/%b/%Y:%H:%M:%S %z

    [PARSER]
        Name        docker
        Format      json
        Time_Key    time
        Time_Format %Y-%m-%dT%H:%M:%S.%L
        Time_Keep   On

    [PARSER]
        Name        docker-daemon
        Format      regex
        Regex       time="(?<time>[^ ]*)" level=(?<level>[^ ]*) msg="(?<msg>[^ ].*)"
        Time_Key    time
        Time_Format %Y-%m-%dT%H:%M:%S.%L
        Time_Keep   On

    [PARSER]
        # http://rubular.com/r/tjUt3Awgg4
        Name cri
        Format regex
        Regex ^(?<time>[^ ]+) (?<stream>stdout|stderr) (?<logtag>[^ ]*) (?<message>.*)$
        Time_Key    time
        Time_Format %Y-%m-%dT%H:%M:%S.%L%z

    [PARSER]
        Name        logfmt
        Format      logfmt

    [PARSER]
        Name        syslog-rfc5424
        Format      regex
        Regex       ^\<(?<pri>[0-9]{1,5})\>1 (?<time>[^ ]+) (?<host>[^ ]+) (?<ident>[^ ]+) (?<pid>[-0-9]+) (?<msgid>[^ ]+) (?<extradata>(\[(.*)\]|-)) (?<message>.+)$
        Time_Key    time
        Time_Format %Y-%m-%dT%H:%M:%S.%L
        Time_Keep   On

    [PARSER]
        Name        syslog-rfc3164-local
        Format      regex
        Regex       ^\<(?<pri>[0-9]+)\>(?<time>[^ ]* {1,2}[^ ]* [^ ]*) (?<ident>[a-zA-Z0-9_\/\.\-]*)(?:\[(?<pid>[0-9]+)\])?(?:[^\:]*\:)? *(?<message>.*)$
        Time_Key    time
        Time_Format %b %d %H:%M:%S
        Time_Keep   On

    [PARSER]
        Name        syslog-rfc3164
        Format      regex
        Regex       /^\<(?<pri>[0-9]+)\>(?<time>[^ ]* {1,2}[^ ]* [^ ]*) (?<host>[^ ]*) (?<ident>[a-zA-Z0-9_\/\.\-]*)(?:\[(?<pid>[0-9]+)\])?(?:[^\:]*\:)? *(?<message>.*)$/
        Time_Key    time
        Time_Format %b %d %H:%M:%S
        Time_Format %Y-%m-%dT%H:%M:%S.%L
        Time_Keep   On

    [PARSER]
        Name    kube-custom
        Format  regex
        Regex   (?<tag>[^.]+)?\.?(?<pod_name>[a-z0-9](?:[-a-z0-9]*[a-z0-9])?(?:\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)_(?<namespace_name>[^_]+)_(?<container_name>.+)-(?<docker_id>[a-z0-9]{64})\.log$
```

DaemonSet:

```
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: fluent-bit
  namespace: vmware-system
  labels:
    k8s-app: fluent-bit
spec:
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      k8s-app: fluent-bit
  template:
    metadata:
      labels:
        k8s-app: fluent-bit
    spec:
      containers:
      - image: projects.registry.vmware.com/tkg/fluent-bit:v1.6.9_vmware.1
        imagePullPolicy: IfNotPresent
        livenessProbe:
          failureThreshold: 3
          httpGet:
            path: /
            port: 2020
            scheme: HTTP
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1
        name: fluent-bit
        ports:
        - containerPort: 2020
          protocol: TCP
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /api/v1/metrics/prometheus
            port: 2020
            scheme: HTTP
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1
        resources:
          limits:
            cpu: 80m
            memory: 200Mi
          requests:
            cpu: 50m
            memory: 100Mi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /var/log
          name: var-log
        - mountPath: /var/log/pods
          name: var-log-pods
        - mountPath: /var/log/containers
          name: var-log-containers
        - mountPath: /var/lib/docker/containers
          name: var-lib-docker-containers
          readOnly: true
        - mountPath: /fluent-bit/etc/
          name: fluent-bit-config
        - mountPath: /run/log
          name: systemd-log
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: fluent-bit
      serviceAccountName: fluent-bit
      terminationGracePeriodSeconds: 10
      tolerations:
      - effect: NoSchedule
        key: node-role.kubernetes.io/master
        operator: Exists
      - effect: NoExecute
        operator: Exists
      - effect: NoSchedule
        operator: Exists
      volumes:
      - hostPath:
          path: /var/log
          type: ""
        name: var-log
      - hostPath:
          path: /var/log/pods
          type: ""
        name: var-log-pods
      - hostPath:
          path: /var/log/containers
          type: ""
        name: var-log-containers
      - hostPath:
          path: /var/lib/docker/containers
          type: ""
        name: var-lib-docker-containers
      - hostPath:
          path: /run/log
          type: ""
        name: systemd-log
      - configMap:
          defaultMode: 420
          name: fluent-bit-config
        name: fluent-bit-config
  updateStrategy:
    rollingUpdate:
      maxUnavailable: 1
    type: RollingUpdate
```

**Expected behavior**
All logs in `/var/log/containers/` should be processed.

**Your Environment**
* Version used: 1.6.9, 1.7.9 and 1.8.9
* Configuration: See above
* Environment name and version Kubernetes Version: 1.20.2
* Server type and version: Single K8s instance
See: https://github.com/vmware-samples/vcenter-event-broker-appliance
* Operating System and version: VMware PhotonOS v4
* Filters and plugins:
    @INCLUDE input-kubernetes.conf
    @INCLUDE input-systemd.conf
    @INCLUDE input-kube-apiserver.conf
    @INCLUDE input-auditd.conf
    @INCLUDE filter-kubernetes.conf
    @INCLUDE filter-record.conf
    @INCLUDE output-syslog.conf

**Additional context**
Running `tail -f` manually from within the system on a specific pod log, which is writing to `stdout`, works.

```
{"log":"10/03/2021 14:47:13 - Handler Processing Completed ...\n","stream":"stdout","time":"2021-10-03T14:47:13.829672574Z"}
{"log":"\n","stream":"stdout","time":"2021-10-03T14:47:13.829772103Z"}
```

Logs which e.g. aren't processed:

```
root@veba-kn [ /var/log/containers ]# ls -rtl                                                                                                                                                                                          
total 376                                                                                                                                                                                                                                                                                                                                                                                                                                     
lrwxrwxrwx 1 root root 100 Sep 13 21:31 antrea-agent-gql5r_kube-system_antrea-agent-8726abf73577f597e15716176cfcdce442b159d00ec12f59e439719d824a9585.log -> /var/log/pods/kube-system_antrea-agent-gql5r_31aa406a-286c-495b-9dcf-e4036c
2a4756/antrea-agent/3.log                                                                                                                                                                                                              
lrwxrwxrwx 1 root root  98 Sep 13 21:31 antrea-agent-gql5r_kube-system_antrea-ovs-3f300f1d7b28c069df1f34cf37ff89be95d69fc3dc4ea0f269b5bd07ce5d56c1.log -> /var/log/pods/kube-system_antrea-agent-gql5r_31aa406a-286c-495b-9dcf-e4036c2a
4756/antrea-ovs/3.log                                                                                                                                                                                                                  
lrwxrwxrwx 1 root root 102 Sep 13 21:31 envoy-89vct_contour-external_shutdown-manager-c8ed97927c25d465f31cce5ab8bd91d02742504f8cf73ad53e493738d0a17f74.log -> /var/log/pods/contour-external_envoy-89vct_1c947a55-2b86-48bd-b442-c6c51e
c2dd3a/shutdown-manager/3.log                                                                                                                                                                                                          
lrwxrwxrwx 1 root root  91 Sep 13 21:31 envoy-89vct_contour-external_envoy-0ea7a33d12105058f74eae9653dd0266ac99ef2ba7f6cb3a3b04a8ec3bc02525.log -> /var/log/pods/contour-external_envoy-89vct_1c947a55-2b86-48bd-b442-c6c51ec2dd3a/envo
y/3.log                                                                                                                                                                                                                                
lrwxrwxrwx 1 root root 104 Sep 13 21:31 contour-5869594b-7jm89_contour-external_contour-803e6591f657fae9539b64ae4f86fa44cce99b409c5f92979c6045cf4b98b52c.log -> /var/log/pods/contour-external_contour-5869594b-7jm89_cc6cf243-7d3f-483
9-91e8-741ab87f6488/contour/3.log                                                                                                                                                                                                      
lrwxrwxrwx 1 root root 106 Sep 13 21:31 contour-5d47766fd8-n24mz_contour-internal_contour-ae34a8ae0b8398da294c5061ec5c0ef1e9be8cb2979f07077e5e9df12f2bab67.log -> /var/log/pods/contour-internal_contour-5d47766fd8-n24mz_a87131ad-d73a
-4371-a47b-dcc410f3b6e4/contour/3.log                                                                                                                                                                                                  
lrwxrwxrwx 1 root root 100 Sep 13 21:31 coredns-74ff55c5b-mjdlr_kube-system_coredns-60bd5f49def85a0ddc929e2c2da5c793a3c6de00cd6a81bdcfdb21f3d4f45129.log -> /var/log/pods/kube-system_coredns-74ff55c5b-mjdlr_7ef260c1-308e-4162-8a84-2
31d560f8023/coredns/3.log
```

I've also tried running the DS in 
```
securityContext:
          privileged: true
```

Similar issues I found but which doesn't provide the solution for this issue:
#3857 
#4014 

Your help would be much appreciated. Thanks

**Comment by rguske** [comment]
I also found #2676...
Running the `DaemonSet` in privileged mode eliminates the `dismissed` message.
```
    spec:
      containers:
      - name: fluent-bit
        image: fluent/fluent-bit:1.8.7
        securityContext:
          privileged: true
``` 

But I still don't see the "real" log output, so what I see when I run `tail` manually on a log, available at my Syslog Server.

**Comment by rguske** [comment]
Regarding the above comment:
We've figured out that the `privileged` was necessary in my config because I used 
```
  input-kubernetes.conf: |
    [INPUT]
        Name              tail
        DB                /var/log/flb_kube.db
```

Means, Fluent Bit was trying to write the file `flb_kube.db` to /var/log/` and it wasn't allowed. Getting rid of the line helped.

**Comment by rguske** [comment]
Fluent Bit was trying to write the file flb_kube.db to /var/log/ and it wasn't allowed. Getting rid of the line helped.

**Comment by zy286938348** [comment]
@rguske I also have this problem, how did you solve it?

**Comment by rguske** [comment]
> @rguske I also have this problem, how did you solve it?

Hi @zy286938348, I have no clue at the moment but I will follow some of the made suggestions here: https://fluent-all.slack.com/archives/C0CTQGHKJ/p1637005220493300 
Please keep me posted if you have more luck with finding the solution for this particular problem. Thanks

**Comment by rguske** [comment]
> 

Hi @zy286938348, 
which container runtime are you using?

**Comment by kc-dot-io** [comment]
@rguske @zy286938348 I am also seeing this issue using containerd as my runtime.
It came up ~16 days after my first deployment of `fluentbit 1.8.1` on a fresh `k8s 1.21` cluster on EKS.
As of now my work around is to simply reinstall fluentbit and then things seem to work again but I don't have a clear RCA.
Even after reinstalling there seems to be specific logs that keep showing up in the `scan_blog add(): dismissed:` error.
It looks like I can't get Slack thread above @rguske - let me know if I can share more info to help figure this out.

**Comment by kc-dot-io** [comment]
I *think* this is the RCA for me: https://github.com/fluent/fluent-bit/issues/3873

When I removed `Ignore_Older` the issues I was seeing went away quite quickly.

**Comment by rguske** [comment]
Hi @kc-dot-io,
I appreciate your comment and help. I wanted to update this issue already last week with my latest observations but last week was a busy one. However, it's interesting that you're mentioning that you are seeing this issue with `containerd` since I figured out, that changing the runtime from `docker` to `containerd` finally provided positive results. Subsequently, I was pretty sure it's due to the old `dockershim`. 
Regarding the use of `Ignore_Older` - I'm sure I've never added it to the `CM` but I will double check this next week.
Again, thanks a lot.

**Comment by teru01** [comment]
I also have this issue. Upgrading Fargate 1.3(using Docker as runtime) to 1.4 (using containerd) did not help.

**Comment by kc-dot-io** [comment]
Circling back to this and I'm no longer certain that `Ignore_Older` is the RCA for me & yes can confirm I'm using containerd.

The issue still appears to occur from time to time and while it's not limite to a single node usually, I have observe the circumstance where one node will work while another isn't. I'm now testing increasing the resources that are being allocated to the fluebt bit replicas since the issue appears to come and go. 

If anyone has any more ideas, I'm happy to troubleshoot it with my setup. Please do ping me in follow ups.

**Comment by rguske** [comment]
Hi all,
it finally worked for our project after replacing `Docker` with `Containerd` and by `apply`ing the following config:
https://github.com/vmware-samples/vcenter-event-broker-appliance/tree/development/files/configs/fluentbit/templates

**Comment by ankitkr** [comment]
Has this been resolved?
This seems to be related to not removal of inode during file removal process i.e. (the inode entry not being removed from files_static, files_event[https://github.com/fluent/fluent-bit/blob/master/plugins/in_tail/tail_config.h#L129, https://github.com/fluent/fluent-bit/blob/master/plugins/in_tail/tail_file.c#1108])

**Comment by edsiper** [comment]
There is a bug in v1.8.13, we have just published the new container image for v1.8.14, please upgrade:

```
fluent/fluent-bit:1.8.14
```

**Comment by danielserrao** [comment]
I'm using version 1.9.0 and I'm getting dismissed logs. This doesn't really seems fixed.

**Comment by andutt** [comment]
Same issue here. Openshift 4 CoreOS and cri-o. Any input regarding this would be great.

**Comment by ramatudaan** [comment]
> @edsiper can you please give us some pointers here to debug this issue. As I have understood
> 
> https://github.com/fluent/fluent-bit/blob/cd92cf36099e69c09b0448e6bc0032bd4ac81ac5/plugins/in_tail/tail_file.c#L861
> 
> this method is returning -1 and the files are getting dismissed. can you please help with some pointers on understanding when does this method return -1.

This got fixed when we started using 1.9.0. @danielserrao

**Comment by andutt** [comment]
@ramatudaan i switched my fluent-bit image to 1.9.0 and my issue now seems to be security/permission related or missconfiguration of templates to match Openshift 4 CoreOS/cri-o. I created and issue here on it: [here](https://github.com/fluent/fluent-bit-kubernetes-logging/issues/97)

**Comment by Lord-Y** [comment]
Hello guys, is there any ETA on this?
Sound like a lot of issues have been opened about this topic and debian packages 1.8.X are not available anymore.
All 1.9.X packages are not fixing the issue.

**Comment by ahvahsky2008** [comment]
Same issue guys, tail logging not works

**Comment by kashirindynatech** [comment]
Have same issue. Tried both output as http to logstash and es plugin to ES

**Comment by sxyandapp** [comment]
问题：使用fluent-bit 的tail插件收集日志时，fluent-bit运行一段时间后，会停止继续收集日志
问题原因：使用multiline.parser进行日志合并时，conn state可能将若干条日志合并为一行，导致tail停止采集
解决思路：
multiline.parser进行日志合并时，受限使用start_state判断是否为开始行，然后使用conn_state判断后续行是否需要合并，因此state_state和conn_state应采用互斥的逻辑，
正确示范：
    rule      "start_state"   "/^(\d{4})(.*)/"     "cont"
    rule      "cont"          "/^(?!\d{4})(.*)/"   "cont"
错误示范：
    rule      "start_state"   "/^(\d{4})(.*)/"     "cont"
    rule      "cont"          "/(.*)/"   "cont"

目前暂时没有遇到tail停止采集的问题

**Comment by Jonathan-w6d** [comment]
Has anyone tried turning `Skip_Long_Lines` to `off` ?
Some of my logs were dismissed, because they were too long I guess, this did the trick

**Comment by zulrang** [comment]
I'm having this issue as well on 1.9.6.  The problem is that I NEED the long lines to be processed.

**Comment by kc-dot-io** [comment]
@Jonathan-w6d yes this helped for me. Also increasing the max buffer size. That said you'll still run into an issue with a memory leak and eventually need to restart the process. I have to do this every 90-120 days.

@zulrang Try increasing your max buffer size to something very large - you'll probably still have to restart the processes every so often, but in theory if they buffer is large enough it should be able to process long lines until it can't again.

**Comment by zulrang** [comment]
> @zulrang Try increasing your max buffer size to something very large - you'll probably still have to restart the processes every so often, but in theory if they buffer is large enough it should be able to process long lines until it can't again.

I have all buffers set to 2M and it has never successfully flushed chunks from one specific node - the one that is logging stack traces (about 8k long).  The other 13 work fine.

**Comment by scostelloppb** [comment]
I'm having the same issue, tried all suggestions in this thread bar catainerd. Tail is dismissing the one and only log I want it to tail. Have tried downgrading to 1.6.10 and upgrading to various versions above up to latest. The problem occurs in every version I now try.

Eks 1.21, docker, Amazon Linux 2 node group, latest helm install from bitnami

**Comment by kc-dot-io** [comment]
> I have all buffers set to 2M and it has never successfully flushed chunks from one specific node - the one that is logging stack traces (about 8k long). The other 13 work fine.

@zulrang did you try going higher? I went all the way up to ~25M using the tail input, k8s filter and skip long lines off before it was stable.

**Comment by zulrang** [comment]
Either there's something seriously wrong with FluentBit or I'm fundamentally misunderstanding how it works If it requires a 25M buffer to process a few 10k long lines

**Comment by Jonathan-w6d** [comment]
@kc-dot-io I have a 100M buffer, Skip_Long_Lines is Off and yesterday I had dismissed lines again, buffer was full, I had to rollout fluent bit. I don't like this.

**Comment by kc-dot-io** [comment]
@Jonathan-w6d Yeah it's definitely not ideal and that sounds quite similar to what I see. I still see the scan_dismissed in the logs, but I feel like that might be a red herring bc I don't seem to be missing anything meaningful once I jack the buffers and turn skip long lines off. From the research I did I think it all comes back to an issue I found that talks about a buffer overflow that was buriedin the issues in this repo. I was thinking about creating a k8s cron that restarts the deployment every 60 days or something. Not ideal but should mitigate. Thoughts on that?

**Comment by Jonathan-w6d** [comment]
@kc-dot-io I guess this would do the trick for you, but I really don't like this idea; with a stable, fully functioning tool you' would never have to do that.
It was my understanding that Fluent bit was a stable tool, but I realise that it's not the case. The only answer in this issue from a staff member was on March 18th...

I think I'm going to have to get rid of fluent bit

**Comment by marratj** [comment]
I have the same issue, both with 1.9.x as well as with 2.0.x. Setting `Max_Buffer_Size` to whatever high value doesn't help at all. Also `Skip_Long_Lines` also doesn't make any difference. 

Even when having no filter in between and simply routing to `stdout`, there's nothing. Just the `scan_blog add(): dismissed` debug message for the affected files and then nothing.

The lines in the affected files are not that long, the longest log line in those files currently is around 500 characters, a simple JSON formatted string, I don't see how that should be a problem.

**Comment by leonardo-albertovich** [comment]
Do you think you could put together a repro case @marratj? That would make it much easier / faster to diagnose and fix.

**Comment by marratj** [comment]
It's a bit special in our case, as we are watching files on a filesystem mounted via CephFS inside a single Fluent-Bit Pod running on Kubernetes.

`Read_from_Head` outputs the initial content just fine, but then no updates to the files are processed.

However, I found a workaround that just works all of a sudden. When I set the `Inotify_Watcher` to `False` in the Tail input, it picks up all new lines written to the log files just fine. Debug logging is still putting out the `scan_blog add(): dismissed` messages for those files, but apart from that it's now working fine with the file stat watcher instead of inotify.

**Comment by leonardo-albertovich** [comment]
I don't know anything about CephFS but it doesn't surprisme that disabling inotify helps. I'll check the error message on Monday to see if it's related to it but I'm curious about the other users too, I wonder if they are using a similar setup.

**Comment by ozhankaraman** [comment]
CephFS is a network mounted filesystem, afaik inotify will not work because kernel is not aware of filesystem changes in a directory, so it can't inform any watch process, maybe it's the issue. If other users are using a network filesystem this could be the case

**Comment by leonardo-albertovich** [comment]
If there are people who are still struggling with this issue, please let us know so we can try to put together a reproduction case or have those of you who are able to run a custom build with additional logging.

From what I could see at first glance, the error string `scan_blog add(): dismissed` is only printed when `flb_tail_file_append` so : 

1. Most of the exit paths of this function produce logs either directly (for most cases) or indirectly [when set_file_pos fails ](https://github.com/fluent/fluent-bit/blob/master/plugins/in_tail/tail_file.c#1119)
2. The only two paths that don't log any errors are the [stat type check](https://github.com/fluent/fluent-bit/blob/master/plugins/in_tail/tail_file.c#L903) and the [already tailed hash check](https://github.com/fluent/fluent-bit/blob/master/plugins/in_tail/tail_file.c#L907).

The stat type check is also performed by the caller [tail_scan_path](https://github.com/fluent/fluent-bit/blob/master/plugins/in_tail/tail_scan_glob.c#L235) and the `stat` instance is the same so I couldn't manage to get the code to take this path even artificially injecting delays and replacing what the link pointed to.

As for the hash check, I took the log failure log lines provided in the original report and since the function that creates the hash uses a combination of the device id and file inode I bruteforced the combinations of device id 1-255 with the inodes he provided but found no collisions (obviously this is an over simplification but it's a decent enough first approach).

**Comment by thusharajayakody1** [comment]
having the same issue. logs are dropping entirely on one node.

**Comment by patrick-stephens** [comment]
There is a possibly related PR to improve collection from logs: https://github.com/fluent/fluent-bit/pull/6552
It doesn't necessarily resolve this (the symptoms are not the message above but general missing logs) and is in current `master` builds so should be in 2.0.9+ (for now you will have to use an non-production-ready `master` build).

**Comment by john-yan** [comment]
Hello all,

I am facing the same issue here using v2.0.8,

Runing fluent-bit on k3s on debian 11. All files are on native sata SSD with EXT4 filesystem.

I was able to identify the root cause to this `readlink` call from https://github.com/fluent/fluent-bit/blob/v2.0.8/plugins/in_tail/tail_file.c#L1543

I wrote a small c prog to prove the behaviour as follows

```c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <linux/limits.h>
#include <errno.h>

int main() {
  char * path = "/var/log/containers/traefik-5758f4779c-g57zc_traefik_traefik-ad1fd81239e26f0cf1713977f526bb8b94be95062830d361ef0ff09ef9c22db3.log";
  FILE * f = fopen(path, "r");
  int fd = fileno(f);
  printf("file descriptor is %d\n", fd);
  char tmp[128];
  char * buf = malloc(PATH_MAX);
  if (!buf) {
    printf("can't malloc\n");
    return 0;
  }
  int ret = snprintf(tmp, sizeof(tmp) - 1, "/proc/%i/fd/%i", getpid(), fd);
  printf("ret=%d, tmp=%s\n", ret, tmp);

  // fluent bit's method of using /proc will fail
  int s = readlink(tmp, buf, PATH_MAX);
  printf("s=%d, err=%d\n", s, errno);

  // use the real path will work
  s = readlink(path, buf, PATH_MAX);
  printf("s=%d, err=%d\n", s, errno);
  return 0;

}
```
Here's the result:
```
root@fluent-bit-4gjpq:/# readlink /var/log/containers/traefik-5758f4779c-g57zc_traefik_traefik-ad1fd81239e26f0cf1713977f526bb8b94be95062830d361ef0ff09ef9c22db3.log
/var/log/pods/traefik_traefik-5758f4779c-g57zc_dd7776dd-b8a0-4acf-a130-ec10427ab19a/traefik/0.log
root@fluent-bit-4gjpq:/# ls -l /var/log/containers/traefik-5758f4779c-g57zc_traefik_traefik-ad1fd81239e26f0cf1713977f526bb8b94be95062830d361ef0ff09ef9c22db3.log
lrwxrwxrwx 1 root root 97 Jan 31 17:36 /var/log/containers/traefik-5758f4779c-g57zc_traefik_traefik-ad1fd81239e26f0cf1713977f526bb8b94be95062830d361ef0ff09ef9c22db3.log -> /var/log/pods/traefik_traefik-5758f4779c-g57zc_dd7776dd-b8a0-4acf-a130-ec10427ab19a/traefik/0.log
root@fluent-bit-4gjpq:/# gcc -o test test.c && ./test 
file descriptor is 3
ret=13, tmp=/proc/91/fd/3
s=-1, err=2
s=97, err=2
```

**Comment by ashutosh-maheshwari** [comment]
Same issue in v2.0.10.
```
[2023/04/04 21:06:11] [debug] [input:tail:tail.3] scan_blog add(): dismissed: /logs/server.out, inode 277548
[2023/04/04 21:06:11] [debug] [input:tail:tail.4] scan_blog add(): dismissed: /logs/server.FATAL, inode 262809
```

Configuration:
```
    Buffer_Chunk_Size 10M
    Buffer_Max_Size 10M
    Skip_Long_Lines On
    Mem_Buf_Limit 30M
```

@edsiper, please let me know which stable version to use for the tail plugin. All the logs are getting dropped.
So many people are looking for a solution to this bug.

**Comment by ashutosh-maheshwari** [comment]
Most versions of Fluent-bit are affected by this bug. I have tested versions 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2.0.6, 2.0.8, and 2.0.10, and they all exhibited the same issue. The only versions that worked for me were `1.2` and `1.3`.
@edsiper @patrick-stephens Can you please review the code for the tail plugin in version 1.3? You might find the bug.

**Comment by neugeeug** [comment]
Hello, 

We are facing the same issue using v2.1.1.
Unfortunately it is a critical blocker for us :(

**Comment by patrick-stephens** [comment]
Is it the same issue identified here? https://github.com/fluent/fluent-bit/issues/4155#issuecomment-1416150129

The gist is a readlink failure with /proc mounts - is that what you have @neugeeug ?

**Comment by ashutosh-maheshwari** [comment]
> > Is it the same issue identified here? [#4155 (comment)](https://github.com/fluent/fluent-bit/issues/4155#issuecomment-1416150129)
> > The gist is a readlink failure with /proc mounts - is that what you have @neugeeug ?
> 
> No, we have just a simple setup, fluent-bit is working as a sidecar in AWS ECS cluster. After some time, we have realized that some long log entries are missing. Debug mode has shown lots of messages:
> 
> ```
> 2023-04-28T17:12:10.580+02:00 | [2023/04/28 15:12:10] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /jcore/logs/config.log, inode 1748378
> 2023-04-28T17:12:10.580+02:00 | [2023/04/28 15:12:10] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /jcore/logs/connector.log, inode 1748381
> ```
> 
> We have been trying to play with memory configuration (Buffer_Max_Size, Buffer_Chunk_Size, Static_Batch_Size,Mem_Buf_Limit), and use an older version 1.8.14. Nothing has helped so far.

@neugeeug Can you try to use v 1.3.7?

**Comment by neugeeug** [comment]
> @neugeeug Can you try to use v 1.3.7?

Well @ashutosh-maheshwari to be honest, I am not sure if we would like to use such old version in production ...

**Comment by brat002** [comment]
Hard to imagine how so important feature could exist for so long. Main feature of the project just doesn't work.

**Comment by leonardo-albertovich** [comment]
Hi @neugeeug, do you think you would be able to reproduce this issue in a reliable way and help me do so myself?

**Comment by neugeeug** [comment]
> Hi @neugeeug, do you think you would be able to reproduce this issue in a reliable way and help me do so myself?

Hi, @leonardo-albertovich we've just finished an extended investigation and it seems that the problem is somewhere else.
Our logs are ending up in Splunk, the problems we were facing were related to a long messages and it seems that there is a limit on the Splunk side, which is most likely rejecting the long batches. It was difficult to find it because logs didn't contain any information about any Splunk problem. It seems also that this part of the log:
`scan_blog add(): dismissed: /jcore/logs/config.log, inode 1863096 `
was not related to the problem. We have just decreased memory buffers, flush period adjusting the values to Splunk limits so the batches approaching Splunk are smaller. It has started working. Thank you all for help.

**Comment by leonardo-albertovich** [comment]
Oh great, I don't know if `out_splunk` has a batch size setting but if it doesn't then it might be a good improvement.

I'd really appreciate it if you were able to validate the status and impact of the change and if appropriate submit an improvement proposal.

On a side note, it seems that is issue is stale and generates confusion. In one hand I'm not in favor of closing issues we are 100% sure have been fixed but with no user input I'm struggling to see the value of keeping this one open so please, if anyone has any input I'd like to hear it.

**Comment by neugeeug** [comment]
> Oh great, I don't know if `out_splunk` has a batch size setting but if it doesn't then it might be a good improvement.

It seems it does not have such setting and I agree, it would help.
Thank you.

**Comment by Jonathan-w6d** [comment]
If it can be of any help, my setup did not include Splunk; just fluent-bit--> opensearch

**Comment by leonardo-albertovich** [comment]
@Jonathan-w6d do you think you could put together a reproduction case so I can analyse it?

**Comment by ashutosh-maheshwari** [comment]
> Actually, considering that the original issue talks about rather old and unsupported versions it might be good to take a step back and re-state everything so we know we are talking about the same thing.
> 
> Would you mind describing your setup and sharing your configuration file? I want to be sure that this issue is still present in the latest release of fluent-bit 2.0 and 2.1.

@leonardo-albertovich On Centos 7.9, install fluent-bit as a systemd service. You can see the above issue. Read my comment - https://github.com/fluent/fluent-bit/issues/4155#issuecomment-1498278814

**Comment by leonardo-albertovich** [comment]
@ashutosh-maheshwari, could you please help putting together a docker based reproduction case with documentation about the deployment process and expectations so it's easier to reproduce the results and ensure we are on the same page?

I know it must be annoying but it would really help improve the pace and ensure the issue is finally resolved.

**Comment by Jonathan-w6d** [comment]
> 

I will try to do so, but tbh I dropped fluentbit out of my stack last July given that noone was helping on the matter.

**Comment by leonardo-albertovich** [comment]
I'm sorry to hear that but I understand you. If you find the time to do it I'd really appreciate it, otherwise, if anyone else wants to step up it'd be great.

**Comment by benjaminhuo** [comment]
> Most versions of Fluent-bit are affected by this bug. I have tested versions 1.2.0, 1.3.0, 1.4.0, 1.5.0, 1.6.0, 1.7.0, 1.8.0, 1.9.0, 2.0.6, 2.0.8, and 2.0.10, and they all exhibited the same issue. **The only versions that worked for me were `1.2.x`, and `1.3.x`.** @edsiper @patrick-stephens Can you please review the code for the tail plugin in version 1.3.7? You might find the bug.

@ashutosh-maheshwari you mean 1.3.6 works for you while 1.3.7 has the same issue?

https://fluentbit.io/announcements/v1.3.7/

![image](https://user-images.githubusercontent.com/18525465/236493746-05fcec96-124e-47be-9489-02d4e2e0b373.png)

**Comment by ashutosh-maheshwari** [comment]
@benjaminhuo It runs on 1.3.7. The problem started post that version.
@Jonathan-w6d What are you using now? I would love to switch.

**Comment by benjaminhuo** [comment]
> @benjaminhuo It runs on 1.3.7. The problem started post that version.
So you mean the changes in 1.3.8 might introduce this bug, right? @ashutosh-maheshwari 
@edsiper @patrick-stephens @leonardo-albertovich , I think this gives us some clues to follow.
![image](https://user-images.githubusercontent.com/18525465/236594051-f161be78-9968-415b-9bc9-9372df43754f.png)

**Comment by leonardo-albertovich** [comment]
Thanks for taking the time to go through the release notes @benjaminhuo, sadly, I don't think that will be enough to get this fixed.

What we need is someone to help put together a reliable reproduction case and I'd like to reiterate that if anyone does I'll take care of having the issue fixed either by doing it myself or having someone from the team work on it.

**Comment by Jonathan-w6d** [comment]
@leonardo-albertovich i can give you the last values.yaml containing the confs that I used before switching to another solution if you'd like ?

**Comment by Jonathan-w6d** [comment]
Here you go 

`kind: DaemonSet

# replicaCount -- Only applicable if kind=Deployment
replicaCount: 1

image:
  repository: cr.fluentbit.io/fluent/fluent-bit
  tag: "1.9.6-debug"
  pullPolicy: Always

#image:
#  repository: gke.gcr.io/gke-metrics-agent
#  tag: "1.8.3-gke.0"
#  pullPolicy: Always

testFramework:
  image:
    repository: busybox
    pullPolicy: Always
    tag: latest

imagePullSecrets: []
nameOverride: ""
fullnameOverride: ""

serviceAccount:
  create: true
  annotations: {}
  name:

rbac:
  create: true
  nodeAccess: true

podSecurityPolicy:
  create: false
  annotations: {}

openShift:
  enabled: false
  securityContextConstraints:
    create: true
    annotations: {}

podSecurityContext: {}

hostNetwork: false
dnsPolicy: ClusterFirst

dnsConfig: {}

hostAliases: []

securityContext: {}

service:
  type: ClusterIP
  port: 2020
  labels: {}
  annotations: {}

serviceMonitor:
  enabled: false

prometheusRule:
  enabled: false

dashboards:
  enabled: false
  labelKey: grafana_dashboard
  annotations: {}
  namespace: ""

lifecycle: {}

livenessProbe:
  httpGet:
    path: /
    port: http

readinessProbe:
  httpGet:
    path: /api/v1/health
    port: http

resources: {}
#   limits:
#     cpu: 100m
#     memory: 128Mi
#   requests:
#     cpu: 100m
#     memory: 128Mi

## only available if kind is Deployment
ingress:
  enabled: false
  className: ""
  annotations: {}
  hosts: []
  extraHosts: []
  tls: []

## only available if kind is Deployment
autoscaling:
  enabled: false
  minReplicas: 1
  maxReplicas: 3
  targetCPUUtilizationPercentage: 75
  customRules: []
  behavior: {}

## only available if kind is Deployment
podDisruptionBudget:
  enabled: false
  annotations: {}
  maxUnavailable: "30%"

nodeSelector: {}

tolerations:
  - operator: Exists
    effect: NoExecute
  - operator: Exists
    effect: NoSchedule
#  - effect: NoSchedule
#    key: dedicated
#    operator: Equal
#    value: appli-strada
#  - effect: NoSchedule
#    key: dedicated
#    operator: Equal
#    value: infra
#  - effect: NoSchedule
#    key: dedicated
#    operator: Equal
#    value: kafka

affinity: {}

labels: {}

annotations: {}

podAnnotations: {}

podLabels: {}

priorityClassName: ""

env: []
#  - name: FOO
#    value: "bar"

# The envWithTpl array below has the same usage as "env", but is using the tpl function to support templatable string.
# This can be useful when you want to pass dynamic values to the Chart using the helm argument "--set <variable>=<value>"
# https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-tpl-function
envWithTpl: []
#  - name: FOO_2
#    value: "{{ .Values.foo2 }}"
#
# foo2: bar2

envFrom: []

extraContainers: []

flush: 5

metricsPort: 2020

extraPorts: []

extraVolumes:
  - name: certs
    secret:
      secretName: tls-for-fluentbit-key-pair

extraVolumeMounts:
  - name: certs
    mountPath: /fluent-bit/etc/certs/ca.crt
    subPath: ca.crt
  - name: certs
    mountPath: /fluent-bit/etc/certs/tls.key
    subPath: tls.key
  - name: certs
    mountPath: /fluent-bit/etc/certs/tls.crt
    subPath: tls.crt

updateStrategy: {}

existingConfigMap: ""

networkPolicy:
  enabled: false

luaScripts: {}

config:
  service: |
    [SERVICE]
        Daemon Off
        Flush {{ .Values.flush }}
        Grace 10
        Log_Level {{ .Values.logLevel }}
        Log_File      /dev/stdout
        Parsers_File custom_parsers.conf
        HTTP_Server On
        HTTP_Listen 0.0.0.0
        HTTP_Port {{ .Values.metricsPort }}
        Storage.path  /var/log/fluentbit
        Storage.sync  normal
        Storage.metrics on

  inputs: |
    [INPUT]
        Name tail
        Path /var/log/containers/*.log
        multiline.parser docker, cri
        Tag kube.*
        Skip_Long_Lines Off
        Mem_Buf_Limit 100MB

  filters: |
    [FILTER]
        Name kubernetes
        Match kube.*
        Merge_Log On
        Buffer_size False

    [FILTER]
        Name        grep
        Match       kube.*

        Exclude     $kubernetes['container_name'] fluent-bit
        Exclude     $kubernetes['container_name'] opensearch
        Exclude     $kubernetes['namespace_name'] logs

    [FILTER]
        Name        modify
        Match       kube.*
        Hard_copy log message

    [FILTER]
        Name          rewrite_tag
        Match         kube.*
        Emitter_Mem_Buf_Limit 100MB

        Rule          $kubernetes['labels']['createdfor'] strada  strada.$TAG false
        Rule          $kubernetes['labels']['app'] gitlab-gitlab  strada.$TAG false
        Rule          $kubernetes['container_image'] strada  strada.$TAG false

    [FILTER]
        Name         parser
        Match_Regex  ^(kube|strada)      
        Key_Name     message
        Reserve_Data True
        Parser       pg
        Parser       glog
        Parser       nginx
        Parser       json
        Parser       kfk

  outputs: |
    [OUTPUT]
        Name opensearch
        Match kube.*
        Host opensearch-cluster-master-headless.logs.svc
        Port 9200
        HTTP_User admin
        HTTP_Passwd admin
        Logstash_Format On
        Logstash_Prefix logk8s
        Logstash_DateFormat %Y.%m.%d
        Retry_Limit False
        Suppress_Type_Name On
        tls On
        tls.verify On
        tls.ca_file /fluent-bit/etc/certs/ca.crt
        tls.crt_file /fluent-bit/etc/certs/tls.crt
        tls.key_file /fluent-bit/etc/certs/tls.key
        Buffer_size False
        Trace_Output On
        Trace_Error On
        Replace_Dots On

    [OUTPUT]
        Name opensearch
        Match strada.*
        Host opensearch-cluster-master-headless.logs.svc
        Port 9200
        HTTP_User admin
        HTTP_Passwd admin
        Logstash_Format On
        Logstash_Prefix logk8s-strada
        Logstash_DateFormat %Y.%m.%d
        Retry_Limit False
        Suppress_Type_Name On
        tls On
        tls.verify On
        tls.ca_file /fluent-bit/etc/certs/ca.crt
        tls.crt_file /fluent-bit/etc/certs/tls.crt
        tls.key_file /fluent-bit/etc/certs/tls.key
        Buffer_size False
        Trace_Output On
        Trace_Error On
        Replace_Dots On

  customParsers: |
    [PARSER]
        Name        cri
        Format      regex
        Regex       ^(?<time>[^ ]+) (?<stream>stdout|stderr) (?<logtag>[^ ]*) (?<log>.*)$
        Time_Key    time
        Time_Format %Y-%m-%dT%H:%M:%S.%L%z

    [PARSER]
        Name        nginx
        Format      regex
        Regex       (?<host>[^ ]*) - (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^\"]*?)(?: +\S*)?)?" (?<code>[^ ]*) (?<size>[^ ]*) "(?<referer>[^\"]*)" "(?<agent>[^\"]*)" (?<request_length>[^ ]*) (?<request_time>[^ ]*) \[(?<proxy_upstream_name>[^ ]*)\] (\[(?<proxy_alternative_upstream_name>[^ ]*)\] )?(?<upstream_addr>[^ ]*) (?<upstream_response_length>[^ ]*) (?<upstream_response_time>[^ ]*) (?<upstream_status>[^ ]*) (?<reg_id>[^ ]*).*$
        Time_Key    time
        Time_Format %d/%b/%Y:%H:%M:%S %z

    [PARSER]
        Name   json
        Format json
        Time_Key time
        Time_Format %d/%b/%Y:%H:%M:%S %z

    [PARSER]
        Name pg
        Format regex
        Regex (?<time>[^ ]+[ ]+[^ ]+[ ]+[^ ]*) \[(?<test>[^\]]*)\] (STATEMENT:\s*(?<statement>[^$].*?)|ERROR:\s*(?<error>[^$].*?)|DETAIL:\s*(?<detail>[^$].*?))$
        Time_Key time
        Time_Format %Y-%m-%d %H:%M:%S.%L %Z

    [PARSER]
        Name        glog
        Format      regex
        Regex       ^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source_file>[^ \]]+)\:(?<source_line>\d+)\]\s(?<message>.*)$
        Time_Key    time
        Time_Format %m%d %H:%M:%S.%L%z

    [PARSER]
        Name        kfk
        Format      regex
        Regex       ^(?<time>[^ ]+[ ]+[^ ]*) \[(?<id>[^\]]*)\] - (?<level>[^ ]*)  \[(?<thread>[^\]]*)\] - (?<message>[^$]*)$
        Time_Key    time
        Format      %Y-%m-%d %H:%M:%S,%L

    [PARSER]
        Name        logfmt
        Format      logfmt

  extraFiles: {}

# The config volume is mounted by default, either to the existingConfigMap value, or the default of "fluent-bit.fullname"
volumeMounts:
  - name: config
    mountPath: /fluent-bit/etc/fluent-bit.conf
    subPath: fluent-bit.conf
  - name: config
    mountPath: /fluent-bit/etc/custom_parsers.conf
    subPath: custom_parsers.conf

daemonSetVolumes:
  - name: varlog
    hostPath:
      path: /var/log
  - name: varlibdockercontainers
    hostPath:
      path: /var/lib/docker/containers
  - name: etcmachineid
    hostPath:
      path: /etc/machine-id
      type: File

daemonSetVolumeMounts:
  - name: varlog
    mountPath: /var/log
  - name: varlibdockercontainers
    mountPath: /var/lib/docker/containers
    readOnly: true
  - name: etcmachineid
    mountPath: /etc/machine-id
    readOnly: true

args: []

command: []

initContainers: []

logLevel: trace`

**Comment by leonardo-albertovich** [comment]
Thank you @Jonathan-w6d, we'll try to have someone from the team reproduce the issue. 
There's no ETA at the moment but I'll send an update as soon as we have some insight.

**Comment by msolters** [comment]
We are encountering this in v2.1.2 in EKS 1.22.
For us it manifests as log files pre-existing when Fluent Bit starts up on a node are tailed and processed.
New files created on that node moving forward yield the `scan_blog add()` debug log message and can't be processed until Fluent Bit is restarted -- so that the files are "pre-existing" by construction.  Not a tenable solution but hopefully a useful symptom diagnostically?

**Comment by benjaminhuo** [comment]
> New files created on that node moving forward yield the `scan_blog add()` debug log message and can't be processed until Fluent Bit is restarted -- so that the files are "pre-existing" by construction.

@msolters That's indeed a good clue to follow, hopefully this can bring some lights for debugging this @leonardo-albertovich

**Comment by leonardo-albertovich** [comment]
Fluent-bit shouldn't have issues detecting new files, at least I never had that happen. Could you please share more information about your context @msolters?

I'm interested in knowing which platform is it, if you are using inotify, how many files are there in the paths where this is happening (or how many glob matchs), how does your configuation file look like and a copy of your log if possible.

Feel free to share things with me privately in slack if you don't feel comfortable doing it in public.

**Comment by wanjunlei** [comment]
[environment.zip](https://github.com/fluent/fluent-bit/files/11980162/environment.zip)

This is some information about the cluster, including the yaml of the Fluent Bit daemonset, the configuration file of Fluent Bit, the version information of k8s and docker, the OS version, and kernel parameters. I hope this helps to reproduce the problem, If you need other information please let me know.

**Comment by benjaminhuo** [comment]
> [environment.zip](https://github.com/fluent/fluent-bit/files/11980162/environment.zip)
> 
> This is some information about the cluster, including the yaml of the Fluent Bit daemonset, the configuration file of Fluent Bit, the version information of k8s and docker, the OS version, and kernel parameters. I hope this helps to reproduce the problem, If you need other information please let me know.

@leonardo-albertovich @edsiper @agup006 @patrick-stephens I hope this can help to narrow down the investigation scope and to locate the root cause

**Comment by IbrahimMCode** [comment]
Same issue using latest version, has this been resolved?

**Comment by emmacz** [comment]
Looks like 'scan_blog add(): dismissed' is generic message, when the root cause can be on different places like in parser etc.

It would be great to add more details in debug mode, which can help to user uncover the issue for example error message coming from parser etc.

Would it be possible?

In general .. where to find the debug information related to parsers processing?

**Comment by benjaminhuo** [comment]
> Looks like 'scan_blog add(): dismissed' is generic message, when the root cause can be on different places like in parser etc.
> 
> It would be great to add more details in debug mode, which can help to user uncover the issue for example error message coming from parser etc.
> 
> Would it be possible?
> 
> In general .. where to find the debug information related to parsers processing?

@leonardo-albertovich what do you think? Is it possible to add more debug messages to help narrow the scope?

**Comment by studyhuang1996** [comment]
![image](https://github.com/fluent/fluent-bit/assets/27665214/e69a6936-8321-4e39-a5ee-757ca975ecc9)

fluent-bit version  2.1.6  has this been resolved?

**Comment by benjaminhuo** [comment]
> ![image](https://user-images.githubusercontent.com/27665214/264525540-e69a6936-8321-4e39-a5ee-757ca975ecc9.png)
> 
> fluent-bit version 2.1.6 has this been resolved?

@studyhuang1996 fluentbit v2.1.6 no longer skips any logs, issue resolved in your env that has this issue before?

**Comment by dntosas** [comment]
FYI team, issue is still present on `v2.1.9` ^

**Comment by dpallagolla** [comment]
I am facing the same issue, one additional observation. The `dismissed:` log seems to appear only in the window when fluent-bit scans for new log files. Once the scan for files is done, logs are forwarded from fluent-bit.

**Comment by praveenkumarp893** [comment]
I tried with latest version of fluentbit (2.1.10) Still facing the same error. Do we have any news on this ?

[2023/10/10 09:28:34] [debug] [input:tail:tail.0] scan_blog add(): dismissed:

 Please see the input config used below.
 
  inputs: |
    [INPUT]
        Name tail
        Path /var/log/containers/*default*.log
        multiline.parser docker, cri, java, multiline-regex-java
        Tag kube.*
        Mem_Buf_Limit 5MB
        Skip_Long_Lines Off

**Comment by studyhuang1996** [comment]
这个问题一直存在，当然也不影响使用，只是不好排查问题，有一定的误解性，但是现在遇到一些问题会出现数据丢失，字段丢失还有一些多行匹配场景不满足，目前打算使用其他工具替换
fluent-bit

Benjamin Huo ***@***.***> 于2023年8月31日周四 14:13写道：

**Comment by omidraha** [comment]
I Have the same issue https://github.com/fluent/helm-charts/issues/415:
```
kubectl logs  -n fluent-bit -f fluent-bit-6341f45a-j8h2x | grep -i volume-test
[2023/10/27 17:04:05] [debug] [input:tail:tail.0] inode=1062251 with offset=2026 appended as /var/log/containers/volume-test_exp_volume-test-2c25a3b1342924cffab9fffb48b2f0a971fcf9f10009e4c382a37bc09075134b.log
[2023/10/27 17:04:05] [debug] [input:tail:tail.0] scan_glob add(): /var/log/containers/volume-test_exp_volume-test-2c25a3b1342924cffab9fffb48b2f0a971fcf9f10009e4c382a37bc09075134b.log, inode 1062251
[2023/10/27 17:04:05] [debug] [input:tail:tail.0] inode=1062251 file=/var/log/containers/volume-test_exp_volume-test-2c25a3b1342924cffab9fffb48b2f0a971fcf9f10009e4c382a37bc09075134b.log promote to TAIL_EVENT
[2023/10/27 17:04:05] [ info] [input:tail:tail.0] inotify_fs_add(): inode=1062251 watch_fd=23 name=/var/log/containers/volume-test_exp_volume-test-2c25a3b1342924cffab9fffb48b2f0a971fcf9f10009e4c382a37bc09075134b.log
[2023/10/27 17:05:04] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /var/log/containers/volume-test_exp_volume-test-2c25a3b1342924cffab9fffb48b2f0a971fcf9f10009e4c382a37bc09075134b.log, inode 1062251
[2023/10/27 17:06:04] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /var/log/containers/volume-test_exp_volume-test-2c25a3b1342924cffab9fffb48b2f0a971fcf9f10009e4c382a37bc09075134b.log, inode 1062251
[2023/10/27 17:07:04] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /var/log/containers/volume-test_exp_volume-test-2c25a3b1342924cffab9fffb48b2f0a971fcf9f10009e4c382a37bc09075134b.log, inode 1062251
[2023/10/27 17:08:04] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /var/log/containers/volume-test_exp_volume-test-2c25a3b1342924cffab9fffb48b2f0a971fcf9f10009e4c382a37bc09075134b.log, inode 1062251
```

**Comment by nulldoot2k** [comment]
Still error: i using version latest and `scan_blog add(): dismissed`
OMG 
![image](https://github.com/fluent/fluent-bit/assets/83489434/49395885-c438-4f8e-a24d-fad8fdee1a8b)

**Comment by nagyzekkyandras** [comment]
same here with fluent/fluent-bit helm chart 0.42.0, and 2.2.1 image version

**Comment by benjaminhuo** [comment]
@kc-dot-io reopen

**Comment by lecaros** [comment]
Hello. If you have this issue, please build your binary using this: https://github.com/lecaros/fluent-bit/tree/master
I've added a few debug messages to narrow down the issue.
Set the `log_level` to `debug`. Run your scenario and share the log file along with your configuration.

**Comment by lecaros** [comment]
I'll gladly reopen the ticket if you can provide the requested indo.

**Comment by lorenzobenvenuti** [comment]
> Hello. If you have this issue, please build your binary using this: https://github.com/lecaros/fluent-bit/tree/master I've added a few debug messages to narrow down the issue. Set the `log_level` to `debug`. Run your scenario and share the log file along with your configuration.

Hi, we had a few scenarios where some logs weren't delivered (using Fluent Bit 2.2.2, "tail" input plugin and "splunk" output plugin). Looking at Fluent Bit logs I've noticed a `scan_blog add(): dismissed` so I thought it was related to the missing events and came here.  I've followed your advice and rebuilt the executable from your fork (BTW, it's missing a couple of `;`), but as far as I can see in our case `dismissed` was a red herring: it's printed even for legit use cases, for example

```
[2024/07/25 11:51:12] [debug] [in_tail] file /var/log/app/access.log already registered
[2024/07/25 11:51:12] [debug] [input:tail:tail.0] scan_blog add(): dismissed: /var/log/app/access.log, inode 16582
1060
```

Basically after FluentBit starts monitoring a file it will always print `dismissed" because it doesn't need to register it again (although TBH the message is misleading, it makes you think that the file is not monitored anymore).

TL,DR: if you came here because you've lost some events and you're seeing `dismissed` in the logs, I'd recommend using the fork to print the root cause (although it'd be great if these info were available in the main development branch) because the actual issue could be unrelated.

Thanks!

**Comment by AstritCepele** [comment]
Can we reopen this issue?

**Comment by srajappa** [comment]
Still seeing this issue even for `3.1.8`

@leonardo-albertovich Logs should paint an appropriate picture of what's wrong. 

An abrupt event out of nowhere will and seeing `dismissed "your file"` sends out panic. And when we see an issue opened several years ago and then coming to a conclusion that fluentbit will likely put out that message is frivolous. 

contemplating switching to a different tool.

**Comment by suhrudkalyan** [comment]
> I _think_ this is the RCA for me: #3873
> 
> When I removed `Ignore_Older` the issues I was seeing went away quite quickly.
> 
> Please tag me if you are able to try this and confirm it works for you too.

**Comment by kc-dot-io** [comment]
Our team just came across this in 4.x.x so it's still an issue but the good news is we stopped seeing the memory leak (that was effectively exhasterbating the issue), since upgrading to 2.x. Here's what we've learned managing this across multiple systems for the last few years...

Thus far our only effective solution has been to continue to increase the `Buffer_Max_Size` while maintaining a reasonable 10-20x ratio to `Buffer_Chunk_Size` to the affected component identified in the `scan_blog add()` error. I think the reason that `Ignore_Older` helped in my situation is that essentially what's happening here is the buffer is being exceeded and the handler isn't doing a great job to surface the reason for this. So in higher volume systems, especially if buffering "older" logs, it exhaust the default buffers quite fast. For us increasing the chunks and buffers to manage the amount of logs coming through the buffer is a tuning proceedure that is a hard requirement in production systems.

IMO, it's a managable work around but it's hard to track down since the error messages are hidden in debug, the buffer defaults are not quite sane for high volume systems, and the error messages don't do a great job to point users in the right direction so this paper trail and document is the key to success IMO. I think the core memory leak bug is fixed and now it's more of an "enhancement" needed for better DX. Probably warrants a new issue, as this one is quite old and confused.

Hope this learning benefits someone else as well.

**Comment by bencorryp** [comment]
@kc-dot-io Thank you for this information! Do you have an example of how high you’re setting Buffer_Max_Size and Buffer_Chunk_Size in production? Hard to tell how big these values should be when fluentbit is recommending ones so low…
